packages/backend/src/peripherals/database/TransactionStatusRepository.ts

```diff
@@ -0,0 +1,122 @@
+import { Hash256, Timestamp } from '@explorer/types'
+import { Knex } from 'knex'
+
+import { Logger } from '../../tools/Logger'
+import { Nullable } from '../../utils/Nullable'
+import { BaseRepository } from './BaseRepository'
+
+interface Row {
+  hash: string
+  sent_at: Nullable<bigint>
+  mined_at: Nullable<bigint>
+  reverted_at: Nullable<bigint>
+  forgotten_at: Nullable<bigint>
+  block_number: Nullable<number>
+  not_found_retries: number
+}
+
+export const NOT_FOUND_RETRIES = 5
+
+export interface Record {
+  hash: Hash256
+  sentAt: Nullable<Timestamp>
+  mined?: {
+    at: Nullable<Timestamp>
+    blockNumber: number
+  }
+  forgottenAt: Nullable<Timestamp>
+  revertedAt: Nullable<Timestamp>
+  notFoundRetries: number
+}
+
+function toRecord(row: Row): Record {
+  const { sent_at, mined_at, block_number } = row
+  const toTimestamp = (value: bigint | null) =>
+    value !== null ? Timestamp(value) : null
+  return {
+    hash: Hash256(row.hash),
+    sentAt: toTimestamp(sent_at),
+    mined:
+      mined_at && block_number
+        ? { at: Timestamp(mined_at), blockNumber: block_number }
+        : undefined,
+    revertedAt: toTimestamp(row.reverted_at),
+    forgottenAt: toTimestamp(row.forgotten_at),
+    notFoundRetries: row.not_found_retries,
+  }
+}
+
+function timestampToBigInt(timestamp?: Nullable<Timestamp>): Nullable<bigint> {
+  return !timestamp ? null : BigInt(timestamp.toString())
+}
+
+function toPartialRow(
+  record: Partial<Record> & { hash: Record['hash'] }
+): Partial<Row> & { hash: Row['hash'] } {
+  const result: Partial<Row> & { hash: Row['hash'] } = {
+    hash: record.hash.toString(),
+    sent_at: record.sentAt && timestampToBigInt(record.sentAt),
```

Comment:
This and the following checks don't work for `0`. In this case maybe it's acceptable, because we won't have timestamps from 1970, but in general it is an error. I'd be happier with explicit undefined checks